src/cmap/connection_pool.ts

```diff
@@ -917,3 +1000,8 @@ export type WithConnectionCallback = (
   connection: Connection | undefined,
   callback: Callback<Connection>
 ) => void;
+
+type AsyncWithConnectionCallback = (
+  error: MongoError | undefined,
+  connection: Connection | undefined
+) => Promise<Connection>;
```

Comment:
NOTE: used to test if implementing `withConnectionAsync` was possible, will be removed